src/XamlMath.Shared/Atoms/BigOperatorAtom.cs

```diff
@@ -23,91 +24,130 @@ private static Box ChangeWidth(Box box, double maxWidth)
-            lowerLimitBox.Shift = -delta / 2;
-            resultBox.Add(lowerLimitBox);
+                limits.LowerLimitBox!.Height), 0, 0));
+            limits.LowerLimitBox.Shift = -boxForBaseAtom.Delta / 2;
+            resultBox.Add(limits.LowerLimitBox);
             resultBox.Add(new StrutBox(0, opSpacing5, 0, 0));
         }
 
         // Adjust height and depth of result box.
-        var baseBoxHeight = baseBox.Height;
+        var baseBoxHeight = boxForBaseAtom.BaseBox.Height;
         var totalHeight = resultBox.Height + resultBox.Depth;
-        if (upperLimitBox != null)
-            baseBoxHeight += opSpacing5 + kern + upperLimitBox.Height + upperLimitBox.Depth;
+        if (limits.UpperLimitBox != null)
+            baseBoxHeight += opSpacing5 + kern + limits.UpperLimitBox.Height + limits.UpperLimitBox.Depth;
         resultBox.Height = baseBoxHeight;
         resultBox.Depth = totalHeight - baseBoxHeight;
 
         return resultBox;
     }
+
+    private readonly record struct FontStyleBundle(ITeXFont TexFont, TexStyle Syle);
+    private static FontStyleBundle CreateFontStyleBundle(TexEnvironment environment) => new(environment.MathFont, environment.Style);
+
+    private static double GetMaxWidth(BoxForBaseAtom boxForBaseAtom, BoxesForUpperAndLowerLimits limits)
+    {
+        var val1 = Math.Max(boxForBaseAtom.BaseBox.Width, limits.UpperLimitBox == null ? 0 : limits.UpperLimitBox.Width);
+        var val2 = limits.LowerLimitBox == null ? 0 : limits.LowerLimitBox.Width;
+        return Math.Max(val1, val2);
+    }
+
+    private readonly record struct BoxesForUpperAndLowerLimits(Box? UpperLimitBox, Box? LowerLimitBox)
+    {
+        internal BoxesForUpperAndLowerLimits AdjustedToMaxWidth(double maxWidth)
+        {
+            Box? newUpperLimitBox = UpperLimitBox == null ? null : ChangeWidth(UpperLimitBox, maxWidth);
+            Box? newLowerLimitBox = LowerLimitBox == null ? null : ChangeWidth(LowerLimitBox, maxWidth);
+            return new(newUpperLimitBox, newLowerLimitBox);
+        }
+    }
+    private BoxesForUpperAndLowerLimits CreateBoxesForUpperAndLowerLimits(TexEnvironment environment)
+    {
+        Box? upperLimitBox = (this.UpperLimitAtom == null)
+            ? null
+            : this.UpperLimitAtom.CreateBox(environment.GetSuperscriptStyle());
+        Box? lowerLimitBox = (this.LowerLimitAtom == null)
+            ? null
+            : this.LowerLimitAtom.CreateBox(environment.GetSubscriptStyle());
+        return new(upperLimitBox, lowerLimitBox);
+    }
+
+    private readonly record struct BoxForBaseAtom(Box BaseBox, double Delta)
+    {
+        internal BoxForBaseAtom AdjustedToMaxWidth(double maxWidth) =>
+            this with { BaseBox = ChangeWidth(BaseBox, maxWidth) };
+    }
+    private BoxForBaseAtom CreateBoxForBaseAtom(TexEnvironment environment)
+    {
+        if (BaseAtom is not SymbolAtom symbolAtom || BaseAtom.Type != TexAtomType.BigOperator)
```

Comment:
_(Minor)_ I'd prefer these branches to be reversed. It is kinda hard to read the inverse condition, while the original "what to do if the base atom is a symbol atom and has a type of `BigOperator`) has lower cognitive load IMO.

What do you think?

Reply:
I must have done that as an intermediate step in some refactoring and then forgotten about it :) but yeah, I agree, inverted logic (i.e. `is not` and `!=`) may be harder to understand. I've now applied your suggestion